### PR TITLE
feat(SqlStores): add default datasource to all SQL stores + tests

### LINKED ISSUE #3337: Global DataSource configuration
# Feature Request

We should provide the ability to set global data source settings for all JDBC-related extensions. If data source settings are not configured for a particular extension, it can default to global settings, if available:

edc.datasource.global.name=
edc.datasource. global.user=
edc.datasource. global.password=
edc.datasource. global.url=

Note we should add a Decision Record to document this.

## Which Areas Would Be Affected?

All JDBC features

## Why Is the Feature Desired?

Ease-of-use

### PR DESCRIPTION
**Pull request by wolf4ood**
## What this PR changes/adds

Add a default datasource as fall back if the specific it's not found in the config

## Why it does that

Ease-of-use

## Further notes

Added also some missing tests on some SQL store extensions

## Linked Issue(s)

Closes #3337

### CODE REVIEW

File: system-tests/e2e-transfer-test/runner/src/test/java/org/eclipse/edc/test/e2e/participant/EndToEndTransferParticipant.java [hunk 1]
```diff
@@ -233,6 +233,21 @@ public Map<String, String> controlPlanePostgresConfiguration() {
         return baseConfiguration;
     }
 
+    public Map<String, String> controlPlanePostgresDefaultConfiguration() {
```
Comment (paullatzelsperger) [inline system-tests/e2e-transfer-test/runner/src/test/java/org/eclipse/edc/test/e2e/participant/EndToEndTransferParticipant.java:236]: wouldn't this make the `controlPlanePostgresConfiguration` in line 204ff obsolete?

File: extensions/control-plane/store/sql/contract-negotiation-store-sql/src/test/java/org/eclipse/edc/connector/store/sql/contractnegotiation/SqlContractNegotiationStoreExtensionTest.java [hunk 2]
```diff
@@ -40,18 +45,25 @@ class SqlContractNegotiationStoreExtensionTest {
 
     @Test
     void initialize(ServiceExtensionContext context, ObjectFactory factory) {
+        var ctx = spy(context);
```
Comment (ndr-brt) [inline extensions/control-plane/store/sql/contract-negotiation-store-sql/src/test/java/org/eclipse/edc/connector/store/sql/contractnegotiation/SqlContractNegotiationStoreExtensionTest.java:48]: not needed, the context it's already wrapped by `spy` by `DependencyInjectionExtension`

File: extensions/control-plane/store/sql/contract-negotiation-store-sql/src/test/java/org/eclipse/edc/connector/store/sql/contractnegotiation/SqlContractNegotiationStoreExtensionTest.java [hunk 3]
```diff
@@ -40,18 +45,25 @@ class SqlContractNegotiationStoreExtensionTest {
...
+        var config = mock(Config.class);
+        when(ctx.getConfig()).thenReturn(config);
+        when(config.getString(DATASOURCE_NAME_SETTING, DataSourceRegistry.DEFAULT_DATASOURCE)).thenReturn("test");
+
         context.registerService(DataSourceRegistry.class, mock(DataSourceRegistry.class));
         context.registerService(TransactionContext.class, mock(TransactionContext.class));
         context.registerService(ContractNegotiationStatements.class, null);
         context.registerService(TypeManager.class, new TypeManager());
 
         extension = factory.constructInstance(SqlContractNegotiationStoreExtension.class);
```
Comment (ndr-brt) [inline extensions/control-plane/store/sql/contract-negotiation-store-sql/src/test/java/org/eclipse/edc/connector/store/sql/contractnegotiation/SqlContractNegotiationStoreExtensionTest.java:58]: Also not needed, the `SqlContractNegotiationStoreExtension` can be injected as method parameter. (same for other tests)

File: system-tests/e2e-transfer-test/runner/src/test/java/org/eclipse/edc/test/e2e/EndToEndTransferPostgresqlDefaultTest.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.HashMap;
+
+import static org.eclipse.edc.test.e2e.PostgresUtil.createDatabase;
+
+@PostgresqlDbIntegrationTest
+class EndToEndTransferPostgresqlDefaultTest extends AbstractEndToEndTransfer {
```
Comment (ndr-brt) [inline system-tests/e2e-transfer-test/runner/src/test/java/org/eclipse/edc/test/e2e/EndToEndTransferPostgresqlDefaultTest.java:29]: I'm not sure about the need to have an e2e tests that verifies the default datasource, I would just change the `EndToEndTransferPostgresqlTest` to make use of the default datasource.

File: system-tests/e2e-transfer-test/runner/src/test/java/org/eclipse/edc/test/e2e/participant/EndToEndTransferParticipant.java (hunk 1 above)
Comment (ndr-brt) [reply system-tests/e2e-transfer-test/runner/src/test/java/org/eclipse/edc/test/e2e/participant/EndToEndTransferParticipant.java:236]: same as I was saying, we could keep just a single postgres e2e transfer test class with the default datasource.

**Review by ndr-brt — COMMENTED** [review]
mainly test nits

File: system-tests/e2e-transfer-test/runner/src/test/java/org/eclipse/edc/test/e2e/participant/EndToEndTransferParticipant.java (hunk 1 above)
Comment (wolf4ood) [reply system-tests/e2e-transfer-test/runner/src/test/java/org/eclipse/edc/test/e2e/participant/EndToEndTransferParticipant.java]: Done

File: extensions/control-plane/store/sql/contract-negotiation-store-sql/src/test/java/org/eclipse/edc/connector/store/sql/contractnegotiation/SqlContractNegotiationStoreExtensionTest.java (hunk 3 above)
Comment (wolf4ood) [reply extensions/control-plane/store/sql/contract-negotiation-store-sql/src/test/java/org/eclipse/edc/connector/store/sql/contractnegotiation/SqlContractNegotiationStoreExtensionTest.java]: I think in this case it's still needed for custom injected services in both test right?

**Review by ndr-brt — APPROVED** [review]

**Pull request merged**